### ISSUE TITLE
[master] Obsolete starter quests

### ISSUE THREAD

**Issue by ArcaneFox** [TrinityCore/TrinityCore#18935]
[//]: # (***************************************************************)
[//]: # (** DON'T DELETE THIS TEMPLATE ELSE YOUR ISSUE WILL BE CLOSED **)
[//]: # (***************************************************************)

**Description:**
This quest was marked obsolete:
http://www.wowhead.com/quest=10068/frost-nova

**Current behaviour:** (Tell us what happens.)
You can accept this quest but you can't complete

**Expected behaviour:** (Tell us what should happen instead.)
This quest shouldn't be offered

**Steps to reproduce the problem:**

1. Create a Blood Elf Mage
2. Level up to 2
3. Go to Julia Sunstriker <Mage Trainer>

**Branch(es):** master

**TC rev. hash/commit:** https://github.com/TrinityCore/TrinityCore/commit/1c00f0b2d405d5f9eee4009ca5810e9721cd36ce

**TDB version:** 703.00 + updates

**Operating system:** Windows 10 64-bit

[//]: # (This template is for problem reports. For other types of report, edit it accordingly.)
[//]: # (If this is a crash report, include the crashlog with https://gist.github.com/)
[//]: # (For fixes containing C++, create a Pull Request.)

**Comment by Keldo** [comment]
**FIX**
```sql
DELETE FROM `creature_queststarter` WHERE id = 15279 AND quest = 10068; -- Quest: Frost Nova
DELETE FROM `creature_questender` WHERE id = 15279 AND quest = 10068; -- Quest: Frost Nova
```

Personally not sure on how to disable the quest in the quest_template table

**Comment by tkrokli** [comment]
Try this (haven't tested it myself yet, but it is the `disables` part):
```sql
DELETE FROM `disables` WHERE `sourceType`= 1 AND `entry`= 15279;
INSERT INTO `disables` (`sourceType`,`entry`,`flags`,`params_0`,`params_1`,`comment`) VALUES
(1, 15279, 0, '', '', 'Deprecated quest: Frost Nova');
```

**Comment by Keldo** [comment]
explain the c_t npcflag please..  Quest are deleted and the quest has been disabled, confirmed but the NPC is question is in fact a quest giver, with the disabled, flags will need to be changed.

**Comment by tkrokli** [comment]
OK, what I meant is this: _if the NPC in question is no longer an actual quest giver_, it is no longer necessary to use the questgiver flag in `creature_template` for this particular creature (https://trinitycore.atlassian.net/wiki/display/tc/creature_template#creature_template-npcflag). As for the SQL to do so:
```sql
UPDATE `creature_template` SET `npcflag` = `npcflag`| ~2 WHERE `entry`= 12345; -- example
```

**Comment by Keldo** [comment]
Good to go

Complete fix below
```sql
DELETE FROM `creature_queststarter` WHERE `id`= 15278 AND `quest`= 8328; -- Quest: Mage Training
DELETE FROM `creature_questender` WHERE `id`= 15279 AND `quest`= 8328; -- Quest: Mage Training
DELETE FROM `creature_queststarter` WHERE `id`= 15279 AND `quest`= 10068; -- Quest: Frost Nova
DELETE FROM `creature_questender` WHERE `id`= 15279 AND `quest`= 10068; -- Quest: Frost Nova
DELETE FROM `disables` WHERE `sourceType`= 1 AND `entry`= 15279;
INSERT INTO `disables` (`sourceType`,`entry`,`flags`,`params_0`,`params_1`,`comment`) VALUES
(1, 15279, 0, '', '', 'Deprecated quest: Frost Nova');
UPDATE `creature_template` SET `npcflag` = `npcflag`| ~2 WHERE `entry`= 15279; -- Quest: Frost Nova
```

**Comment by tkrokli** [comment]
Sorry for creating confusion, I read up on and corrected the SQL queries above to make them correct again.

**Comment by Keldo** [comment]
All good, now I can start fixing the rest of the Known Class Quest that are no longer needed and maybe fix some broken quest. Hit me up on IRC if needed.

**Comment by Keldo** [comment]
**Blood Elf Class Quest Fixes**
```sql
/*
	Blood Elf
	Start Area
	Class Quest that are obsolete
*/

-- MAGE
-- Julia Sunstriker <Mage Trainer> is no longer a quest giver
UPDATE `creature_template` SET `npcflag` = `npcflag`& ~2 WHERE `entry`= 15279;

-- Quest: Mage Training (8328) removed in patch 7.0.3
DELETE FROM `creature_queststarter` WHERE `id`= 15278 AND `quest`= 8328;
DELETE FROM `creature_questender` WHERE `id`= 15279 AND `quest`= 8328;

-- Quest: Frost Nova (10068) removed in patch 7.0.3
DELETE FROM `creature_queststarter` WHERE `id`= 15279 AND `quest`= 10068;
DELETE FROM `creature_questender` WHERE `id`= 15279 AND `quest`= 10068;

DELETE FROM `disables` WHERE `sourceType`= 1 AND `entry` IN (8328,10068);
INSERT INTO `disables` (`sourceType`,`entry`,`flags`,`params_0`,`params_1`,`comment`) VALUES
(1,  8328, 0, '', '', 'Quest removed in patch 7.0.3: Mage Training'),
(1, 10068, 0, '', '', 'Quest removed in patch 7.0.3: Frost Nova');

-- HUNTER
-- Ranger Sallina <Hunter Trainer> is no longer a quest giver
UPDATE `creature_template` SET `npcflag` = `npcflag`& ~2 WHERE `entry`= 15513;

-- Quest: Hunter Training (9393) removed in patch 7.0.3
DELETE FROM `creature_queststarter` WHERE `id`= 15278 AND `quest`= 9393;
DELETE FROM `creature_questender` WHERE `id`= 15513 AND `quest`= 9393;

-- Quest: Steady Shot (10070) removed in patch 7.0.3
DELETE FROM `creature_queststarter` WHERE `id`= 15513 AND `quest`= 10070;
DELETE FROM `creature_questender` WHERE `id`= 15295 AND `quest`= 10070;

DELETE FROM `disables` WHERE `sourceType`= 1 AND `entry` IN (9393,10070);
INSERT INTO `disables` (`sourceType`,`entry`,`flags`,`params_0`,`params_1`,`comment`) VALUES
(1,  9393, 0, '', '', 'Quest removed in patch 7.0.3: Hunter Training'),
(1, 10070, 0, '', '', 'Quest removed in patch 7.0.3: Steady Shot');

-- ROGUE
-- Pathstalker Kariel <Rogue Trainer> is no longer a quest giver
UPDATE `creature_template` SET `npcflag` = `npcflag`& ~2 WHERE `entry`= 15285;

-- Quest: Rogue Training (9392) removed in patch 7.0.3
DELETE FROM `creature_queststarter` WHERE `id`= 15278 AND `quest`= 9392;
DELETE FROM `creature_questender` WHERE `id`= 15285 AND `quest`= 9392;

-- Quest: Evisceration (9393) removed in patch 7.0.3
DELETE FROM `creature_queststarter` WHERE `id`= 15285 AND `quest`= 9393;
DELETE FROM `creature_questender` WHERE `id`= 15295 AND `quest`= 9393;

DELETE FROM `disables` WHERE `sourceType`= 1 AND `entry` IN (9392,10071);
INSERT INTO `disables` (`sourceType`,`entry`,`flags`,`params_0`,`params_1`,`comment`) VALUES
(1,  9392, 0, '', '', 'Quest removed in patch 7.0.3: Rogue Training'),
(1, 10071, 0, '', '', 'Quest removed in patch 7.0.3: Evisceration');

-- WARRIOR
-- Delios Silverblade <Warrior Trainer> is no longer a quest giver
UPDATE `creature_template` SET `npcflag` = `npcflag`& ~2 WHERE `entry`= 43010;

-- Quest: Warrior Training (9289) removed in patch 7.0.3
DELETE FROM `creature_queststarter` WHERE `id`= 15278 AND `quest`= 9289;
DELETE FROM `creature_questender` WHERE `id`= 43010 AND `quest`= 9289;

-- Quest: Charge (27091) removed in patch 7.0.3
DELETE FROM `creature_queststarter` WHERE `id`= 43010 AND `quest`= 27091;
DELETE FROM `creature_questender` WHERE `id`= 43010 AND `quest`= 27091;

DELETE FROM `disables` WHERE `sourceType`= 1 AND `entry` IN (9389,27091);
INSERT INTO `disables` (`sourceType`,`entry`,`flags`,`params_0`,`params_1`,`comment`) VALUES
(1,  9383, 0, '', '', 'Quest removed in patch 7.0.3: Warrior Training'),
(1, 27091, 0, '', '', 'Quest removed in patch 7.0.3: Charge');

-- PRIEST
-- Matron Arena <Priest Trainer> is no longer a quest giver
UPDATE `creature_template` SET `npcflag` = `npcflag`& ~2 WHERE `entry`= 15284;

-- Quest: Priest Training (8564) removed in patch 7.0.3
DELETE FROM `creature_queststarter` WHERE `id`= 15278 AND `quest`= 8564;
DELETE FROM `creature_questender` WHERE `id`= 15284 AND `quest`= 8564;

-- Quest: Learning the Word (10072) removed in patch 7.0.3
DELETE FROM `creature_queststarter` WHERE `id`= 15284 AND `quest`= 10072;
DELETE FROM `creature_questender` WHERE `id`= 15295 AND `quest`= 10072;

DELETE FROM `disables` WHERE `sourceType`= 1 AND `entry` IN (8564,10072);
INSERT INTO `disables` (`sourceType`,`entry`,`flags`,`params_0`,`params_1`,`comment`) VALUES
(1,  8564, 0, '', '', 'Quest removed in patch 7.0.3: Priest Training'),
(1, 10072, 0, '', '', 'Quest removed in patch 7.0.3: Learning the Word');

-- PALADIN
-- Jesthenis Sunstriker <Paladin Trainer> is no longer a quest giver
UPDATE `creature_template` SET `npcflag` = `npcflag`& ~2 WHERE `entry`= 15280;

-- Quest: Paladin Training (9676) removed in patch 7.0.3
DELETE FROM `creature_queststarter` WHERE `id`= 15278 AND `quest`= 9676;
DELETE FROM `creature_questender` WHERE `id`= 15280 AND `quest`= 9676;

-- Quest: Ways of the Light (10069) removed in patch 7.0.3
DELETE FROM `creature_queststarter` WHERE `id`= 15280 AND `quest`= 10069;
DELETE FROM `creature_questender` WHERE `id`= 15295 AND `quest`= 10069;

DELETE FROM `disables` WHERE `sourceType`= 1 AND `entry` IN (9676,10069);
INSERT INTO `disables` (`sourceType`,`entry`,`flags`,`params_0`,`params_1`,`comment`) VALUES
(1,  9675, 0, '', '', 'Quest removed in patch 7.0.3: Paladin Training'),
(1, 10069, 0, '', '', 'Quest removed in patch 7.0.3: Ways of the Light');

-- WARLOCK
-- Summoner Teli'Larien <Warlock Trainer> is no longer a quest giver
UPDATE `creature_template` SET `npcflag` = `npcflag`& ~2 WHERE `entry`= 15283;

-- Quest: Warlock Training (8563) removed in patch 7.0.3
DELETE FROM `creature_queststarter` WHERE `id`= 15278 AND `quest`= 8563;
DELETE FROM `creature_questender` WHERE `id`= 15283 AND `quest`= 8563;

-- Quest: Curruption (10073) removed in patch 7.0.3
DELETE FROM `creature_queststarter` WHERE `id`= 15283 AND `quest`= 10073;
DELETE FROM `creature_questender` WHERE `id`= 15295 AND `quest`= 10073;

DELETE FROM `disables` WHERE `sourceType`= 1 AND `entry` IN (8563,10073);
INSERT INTO `disables` (`sourceType`,`entry`,`flags`,`params_0`,`params_1`,`comment`) VALUES
(1,  8563, 0, '', '', 'Quest removed in patch 7.0.3: Warlock Training'),
(1, 10073, 0, '', '', 'Quest removed in patch 7.0.3: Curruption');

-- MONK
-- Pao <Monk Trainer> is no longer a quest giver
UPDATE `creature_template` SET `npcflag` = `npcflag`& ~2 WHERE `entry`= 63332;

-- Quest: Monk Training (31170) removed in patch 7.0.3
DELETE FROM `creature_queststarter` WHERE `id`= 15278 AND `quest`= 31170;
DELETE FROM `creature_questender` WHERE `id`= 63332 AND `quest`= 31170;

-- Quest: Tiger Palm (31171) removed in patch 7.0.3
DELETE FROM `creature_queststarter` WHERE `id`= 63332 AND `quest`= 31171;
DELETE FROM `creature_questender` WHERE `id`= 63332 AND `quest`= 31171;

DELETE FROM `disables` WHERE `sourceType`= 1 AND `entry` IN (31170,31171);
INSERT INTO `disables` (`sourceType`,`entry`,`flags`,`params_0`,`params_1`,`comment`) VALUES
(1,  31170, 0, '', '', 'Quest removed in patch 7.0.3: Monk Training'),
(1, 31171, 0, '', '', 'Quest removed in patch 7.0.3: Tiger Palm');
```

**Comment by tkrokli** [comment]
Edited the SQL above and made the following changes:
- aligned `WHERE id =` in the `creature_queststarter` / `creature_questender` lines for readability
- replaced the `/* */` comment style with `-- ` for backwards compatibility with SQL GUI editors

**Comment by Keldo** [comment]
Working on The Undead Class Start Obsoletes today even though there are still 7 quest missing, in the DB, not in disables but not available

**Comment by tkrokli** [comment]
Not quite sure where to look for the reason some of the quests aren't available besides the "usual suspects".
Maybe we should try looking for non-working spells or missing DB conditions as possible causes?

**Comment by Aokromes** [comment]
put all the deletes on a single group and the same for inserts and updates :)

**Comment by tkrokli** [comment]
Thanks, I am on it. Looks like there are a couple of typos, need to sort the lines & correct the quest IDs.

**Comment by Keldo** [comment]
**UNDEAD**
```sql
-- Undead start area obsoletes

-- HUNTER
-- Xavier the Huntsman <Hunter Trainer> is no longer a quest giver
UPDATE `creature_template` SET `npcflag` = `npcflag`& ~2 WHERE `entry`= 38911;

-- Quest: Trail-Worn Scroll (24962) was removed in patch 7.0.3
DELETE FROM `creature_queststarter` WHERE `id`= 1569 AND `quest`= 24962;
DELETE FROM `creature_questender` WHERE `id`= 38911 AND `quest`= 24962;

-- Quest: The Thrill of the Hunt (24964) removed in patch 7.0.3
DELETE FROM `creature_queststarter` WHERE `id`= 38911 AND `quest`= 24964;
DELETE FROM `creature_questender` WHERE `id`= 38911 AND `quest`= 24964;

DELETE FROM `disables` WHERE `sourceType`= 1 AND `entry` IN (24962,24964);
INSERT INTO `disables` (`sourceType`,`entry`,`flags`,`params_0`,`params_1`,`comment`) VALUES
(1,  24962, 0, '', '', 'Quest removed in patch 7.0.3: Trail-Worn Scroll '),
(1,  24964, 0, '', '', 'Quest removed in patch 7.0.3: The Thrill of the Hunt ');

-- WARRIOR
-- Dannel Stern <Warrior Trainer> is no longer a quest giver
UPDATE `creature_template` SET `npcflag` = `npcflag`& ~2 WHERE `entry`= 2119;

-- Quest: Simple Scroll (3095) was removed in patch 7.0.3
DELETE FROM `creature_queststarter` WHERE `id`= 1569 AND `quest`= 3095;
DELETE FROM `creature_questender` WHERE `id`= 2119 AND `quest`= 3095;

-- Quest: Charging into Battle (24969) removed in patch 7.0.3
DELETE FROM `creature_queststarter` WHERE `id`= 2119 AND `quest`= 24969;
DELETE FROM `creature_questender` WHERE `id`= 2119 AND `quest`= 24969;

DELETE FROM `disables` WHERE `sourceType`= 1 AND `entry` IN (3095,24969);
INSERT INTO `disables` (`sourceType`,`entry`,`flags`,`params_0`,`params_1`,`comment`) VALUES
(1,  3095, 0, '', '', 'Quest removed in patch 7.0.3: Simple Scroll '),
(1,  24969, 0, '', '', 'Quest removed in patch 7.0.3: Charging into Battle ');

-- ROGUE
-- David Trias <Rogue Trainer> is no longer a quest giver
UPDATE `creature_template` SET `npcflag` = `npcflag`& ~2 WHERE `entry`= 2122;

-- Quest: Encrypted Scroll (3096) was removed in patch 7.0.3
DELETE FROM `creature_queststarter` WHERE `id`= 1569 AND `quest`= 3096;
DELETE FROM `creature_questender` WHERE `id`= 2122 AND `quest`= 3096;

-- Quest: Stab! (24967) removed in patch 7.0.3
DELETE FROM `creature_queststarter` WHERE `id`= 2122 AND `quest`= 24967;
DELETE FROM `creature_questender` WHERE `id`= 2122 AND `quest`= 24967;

DELETE FROM `disables` WHERE `sourceType`= 1 AND `entry` IN (3096,24967);
INSERT INTO `disables` (`sourceType`,`entry`,`flags`,`params_0`,`params_1`,`comment`) VALUES
(1,  3096, 0, '', '', 'Quest removed in patch 7.0.3: Encrypted Scroll '),
(1,  24967, 0, '', '', 'Quest removed in patch 7.0.3: Stab! ');

-- PRIEST
-- Dark Cleric Duesten <Priest Trainer> is no longer a quest giver
UPDATE `creature_template` SET `npcflag` = `npcflag`& ~2 WHERE `entry`= 2123;

-- Quest: Hallowed Scroll (3097) was removed in patch 7.0.3
DELETE FROM `creature_queststarter` WHERE `id`= 1569 AND `quest`= 3097;
DELETE FROM `creature_questender` WHERE `id`= 2122 AND `quest`= 3097;

-- Quest: Of Light and Shadows (24966) removed in patch 7.0.3
DELETE FROM `creature_queststarter` WHERE `id`= 2123 AND `quest`= 24966;
DELETE FROM `creature_questender` WHERE `id`= 2123 AND `quest`= 24966;

DELETE FROM `disables` WHERE `sourceType`= 1 AND `entry` IN (3097,24966);
INSERT INTO `disables` (`sourceType`,`entry`,`flags`,`params_0`,`params_1`,`comment`) VALUES
(1,  3097, 0, '', '', 'Quest removed in patch 7.0.3: Hallowed Scroll '),
(1,  24966, 0, '', '', 'Quest removed in patch 7.0.3: Of Light and Shadows ');

-- MAGE
-- Isabella <Mage Trainer> is no longer a quest giver
UPDATE `creature_template` SET `npcflag` = `npcflag`& ~2 WHERE `entry`= 2124;

-- Quest: Glyphic Scroll (3098) was removed in patch 7.0.3
DELETE FROM `creature_queststarter` WHERE `id`= 1569 AND `quest`= 3098;
DELETE FROM `creature_questender` WHERE `id`= 2124 AND `quest`= 3098;

-- Quest: Magic Training (24965) removed in patch 7.0.3
DELETE FROM `creature_queststarter` WHERE `id`= 2124 AND `quest`= 24965;
DELETE FROM `creature_questender` WHERE `id`= 2124 AND `quest`= 24965;

DELETE FROM `disables` WHERE `sourceType`= 1 AND `entry` IN (3098,24965);
INSERT INTO `disables` (`sourceType`,`entry`,`flags`,`params_0`,`params_1`,`comment`) VALUES
(1,  3098, 0, '', '', 'Quest removed in patch 7.0.3: Glyphic Scroll '),
(1,  24965, 0, '', '', 'Quest removed in patch 7.0.3: Magic Training ');

-- WARLOCK
-- Maximillion <Warlock Trainer> is no longer a quest giver
UPDATE `creature_template` SET `npcflag` = `npcflag`& ~2 WHERE `entry`= 2126;

-- Quest: Tainted Scroll (3099) was removed in patch 7.0.3
DELETE FROM `creature_queststarter` WHERE `id`= 1569 AND `quest`= 3099;
DELETE FROM `creature_questender` WHERE `id`= 2126 AND `quest`= 3099;

-- Quest: Dark Deeds (24968) removed in patch 7.0.3
DELETE FROM `creature_queststarter` WHERE `id`= 2126 AND `quest`= 24968;
DELETE FROM `creature_questender` WHERE `id`= 2126 AND `quest`= 24968;

DELETE FROM `disables` WHERE `sourceType`= 1 AND `entry` IN (3099,24968);
INSERT INTO `disables` (`sourceType`,`entry`,`flags`,`params_0`,`params_1`,`comment`) VALUES
(1,  3099, 0, '', '', 'Quest removed in patch 7.0.3: Tainted Scroll '),
(1,  24968, 0, '', '', 'Quest removed in patch 7.0.3: Dark Deeds ');
```

**Comment by Keldo** [comment]
Sorry about that pull request yall. I will just keep updating here

**Comment by tkrokli** [comment]
Updated "Blood Elf Class Quest Fixes" in https://github.com/TrinityCore/TrinityCore/issues/18935#issuecomment-274882435

Please note the changed format, in addition to the fact that I had to check **every single quest** and correct the typos and also remove some alliance quests/quest givers.  [Well Watcher Solanian (15295)](http://www.wowhead.com/npc=15295/well-watcher-solanian) entries also had to be rewritten based on contents from my current TDB as of rev. c36f831ab714 2017-01-24 00:08:12 +0100.

**Comment by Keldo** [comment]
umm ok, either way it works, almost done with Orc now

**Comment by tkrokli** [comment]
Sure, it would work, it just would have been more than blood elf quests and some empty DB entries.

**Comment by Keldo** [comment]
I'm saying either format your fancy one or my generic one

**Comment by Keldo** [comment]
**ORC**
```sql
-- ORC

-- UPDATES
-- Frang <Warrior Trainer> is no longer a quest giver
UPDATE `creature_template` SET `npcflag` = `npcflag`& ~2 WHERE `entry`= 3153;
-- Karranisha <Hunter Trainer> is no longer a quest giver
UPDATE `creature_template` SET `npcflag` = `npcflag`& ~2 WHERE `entry`= 39214;
-- Rwag <Rogue Trainer> is no longer a quest giver
UPDATE `creature_template` SET `npcflag` = `npcflag`& ~2 WHERE `entry`= 3155;
-- Shikrik <Shaman Trainer> is no longer a quest giver
UPDATE `creature_template` SET `npcflag` = `npcflag`& ~2 WHERE `entry`= 3157;

-- DELETE

-- Quest: Simple Parchment (2383) removed in patch 7.0.3
DELETE FROM `creature_queststarter` WHERE `id`= 3143 AND `quest`= 2383;
DELETE FROM `creature_questender` WHERE `id`= 3153 AND `quest`= 2383;

-- Quest: Charge (25147) removed in patch 7.0.3
DELETE FROM `creature_queststarter` WHERE `id`= 3153 AND `quest`= 25147;
DELETE FROM `creature_questender` WHERE `id`= 3153 AND `quest`= 25147;
DELETE FROM `disables` WHERE `sourceType`= 1 AND `entry` IN (2383,25147);

-- Quest: Etched Parchment (3087) removed in patch 7.0.3
DELETE FROM `creature_queststarter` WHERE `id`= 3143 AND `quest`= 3087;
DELETE FROM `creature_questender` WHERE `id`= 39214 AND `quest`= 3087;

-- Quest: Steady Shot (25139) removed in patch 7.0.3
DELETE FROM `creature_queststarter` WHERE `id`= 39214 AND `quest`= 25139;
DELETE FROM `creature_questender` WHERE `id`= 39214 AND `quest`= 25139;
DELETE FROM `disables` WHERE `sourceType`= 1 AND `entry` IN (3087,25139);

-- Quest: Encrypted Parchment (3088) removed in patch 7.0.3
DELETE FROM `creature_queststarter` WHERE `id`= 3143 AND `quest`= 3088;
DELETE FROM `creature_questender` WHERE `id`= 3155 AND `quest`= 3088;

-- Quest: Eviscerate (25141) removed in patch 7.0.3
DELETE FROM `creature_queststarter` WHERE `id`= 3155 AND `quest`= 25141;
DELETE FROM `creature_questender` WHERE `id`= 3155 AND `quest`= 25141;
DELETE FROM `disables` WHERE `sourceType`= 1 AND `entry` IN (3087,25139);

-- Quest: Rune-Inscribed Parchment (3089) removed in patch 7.0.3
DELETE FROM `creature_queststarter` WHERE `id`= 3143 AND `quest`= 3089;
DELETE FROM `creature_questender` WHERE `id`= 3157 AND `quest`= 3089;
-- Quest: Primal Strike (25143) removed in patch 7.0.3
DELETE FROM `creature_queststarter` WHERE `id`= 3157 AND `quest`= 25143;
DELETE FROM `creature_questender` WHERE `id`= 3157 AND `quest`= 25143;
DELETE FROM `disables` WHERE `sourceType`= 1 AND `entry` IN (3089,25143);

-- INSERT

INSERT INTO `disables` (`sourceType`,`entry`,`flags`,`params_0`,`params_1`,`comment`) VALUES
(1,  2383, 0, '', '', 'Quest removed in patch 7.0.3: Simple Parchment'),
(1, 25147, 0, '', '', 'Quest removed in patch 7.0.3: Charge');
(1,  3087, 0, '', '', 'Quest removed in patch 7.0.3: Etched Parchment'),
(1, 25139, 0, '', '', 'Quest removed in patch 7.0.3: Steady Shot'),
(1,  3088, 0, '', '', 'Quest removed in patch 7.0.3: Encrypted Parchment'),
(1, 25141, 0, '', '', 'Quest removed in patch 7.0.3: Eviscerate'),
(1,  3089, 0, '', '', 'Quest removed in patch 7.0.3: Rune-Inscribed Parchment'),
(1, 25143, 0, '', '', 'Quest removed in patch 7.0.3: Primal Strike');
```

**Comment by Keldo** [comment]
- [x] Blood Elf
- [x] Undead
- [x] Orc
- [ ] Goblin
- [ ] Troll
- [ ] Human
- [ ] Draenei
- [ ] Gnome
- [ ] Dwarf

**Comment by tkrokli** [comment]
Formatted the SQL for Deathknell (Undead starting area), but only verified the NPCs, not all the quest info.

**Comment by Keldo** [comment]
**Human Startng Area (Northshire Valley)**
```sql
-- Quest and quest givers removed in patch 7.0.3

-- The Following NPC's are no longer quest givers
UPDATE `creature_template` SET `npcflag` = `npcflag`& ~2 WHERE `entry` IN
(911,  -- Llane Beshere <Warrior Trainer>
925,   -- Bother Sammuel <Paladin Trainer>
375,   -- Priestess Anetta <Priest Trainer>
43278, -- Ashley Bank <Hunter Trainer>
459,   -- Drusilla La Salle <Warlock Trainer>
198);  -- Khelden Bremen <Mage Trainer>

-- Marshal Mcbride (197) no longer starts these quests:
DELETE FROM `creature_queststarter` WHERE `id`= 192 AND `quest` IN
(3100,  -- Quest:Simple Letter
 3101,  -- Quest:Consecrated Letter
 26910, -- Quest:Etched Letter
 3102,  -- Quest:Encrypted Letter
 3103,  -- Quest:Hallowed Letter
 3104,  -- Quest:Glyphic Letter
 3105); -- Quest:Tainted Letter
 
 -- These NPCs no longer start the following obsolete quests:
DELETE FROM `creature_queststarter` WHERE `id`= 925 AND `quest`= 26918; -- Brother Sammuel
DELETE FROM `creature_queststarter` WHERE `id`= 911 AND `quest`= 26913; -- Llane Beshere
DELETE FROM `creature_queststarter` WHERE `id`= 43278 AND `quest` = 26917; -- Ashley Bank
DELETE FROM `creature_queststarter` WHERE `id`= 915 AND `quest` = 26915; -- Jorik Kerridan
DELETE FROM `creature_queststarter` WHERE `id`= 375 AND `quest` = 26919; -- Priestess Anetta
DELETE FROM `creature_queststarter` WHERE `id`= 459 AND `quest` = 26914; -- Drusilla La Salle

-- These NPCs no longer end the following obsolete quests:
DELETE FROM `creature_questender` WHERE `id`= 925 AND `quest` = 26918; -- Brother Sammuel
DELETE FROM `creature_questender` WHERE `id`= 911 AND `quest` IN (3100,26913); -- Llane Beshere
DELETE FROM `creature_questender` WHERE `id`= 43278 AND `quest` = 26917; -- Ashley Bank
DELETE FROM `creature_questender` WHERE `id`= 915 AND `quest` IN (3102,26915); -- Jorik Kerridan
DELETE FROM `creature_questender` WHERE `id`= 375 AND `quest` IN (3103,26919); -- Priestess Anetta
DELETE FROM `creature_questender` WHERE `id`= 375 AND `quest` IN (3105,26919); -- Priestess Anetta

-- These quests are obsolete and need to be disabled:
DELETE FROM `disables` WHERE `sourceType`= 1 AND `entry` IN (3100,3101,26918,26913,26917,3102,26915,3103,26919,3104,3105,26914);
INSERT INTO `disables` (`sourceType`,`entry`,`flags`,`params_0`,`params_1`,`comment`) VALUES
(1,  3100, 0, '', '', 'Obsolete quest: Simple Letter'),
(1,  3101, 0, '', '', 'Obsolete quest: Consecrated Letter'),
(1,  26918, 0, '', '', 'Obsolete quest: The Power of the Light'),
(1,  26913, 0, '', '', 'Obsolete quest: Charging into Battle'),
(1,  26910, 0, '', '', 'Obsolete quest: Etched Letter'),
(1,  26917, 0, '', '', 'Obsolete quest: The Hunter\'s Path'),
(1,  3102, 0, '', '', 'Obsolete quest: Encrypted Letter'),
(1,  26915, 0, '', '', 'Obsolete quest: The Deepest Cut'),
(1,  3103, 0, '', '', 'Obsolete quest: Hallowed Letter'),
(1,  26919, 0, '', '', 'Obsolete quest: Learning the Word'),
(1,  3104, 0, '', '', 'Obsolete quest: Glyphic Letter'),
(1,  3105, 0, '', '', 'Obsolete quest: Tainted Letter'),
(1,  26914, 0, '', '', 'Obsolete quest: Corruption');
```

**Comment by Keldo** [comment]
Added Ting <Monk Trainer> to the Undead SQL

**Comment by Keldo** [comment]
https://github.com/TrinityCore/TrinityCore/issues/18926 90% Complete, 2 Minor issues to address

**Comment by Keldo** [comment]
#18926 100% Complete

**Comment by Keldo** [comment]
**WIP - Starting the Nigh Elf Start Area**
```sql
-- Quest and quest givers removed in patch 7.0.3

-- The Following NPC's are no longer quest givers
UPDATE `creature_template` SET `npcflag` = `npcflag`& ~2 WHERE `entry` IN
(3596);  -- Ayanna Everstride <Hunter Trainer>);

-- Illithaine (2079) no longer starts these quests:
DELETE FROM `creature_queststarter` WHERE `id`= 2079 AND `quest` IN
(3117);  -- Quest:Etched Sigil

 
  -- These NPCs no longer start the following obsolete quests:
DELETE FROM `creature_queststarter` WHERE `id`= 3596 AND `quest`= 26947; -- A Woodsman's Training

-- These NPCs no longer end the following obsolete quests:
DELETE FROM `creature_questender` WHERE `id`= 3596 AND `quest` IN (3117,26947); -- Ayanna Everstride

-- These quests are obsolete and need to be disabled:
DELETE FROM `disables` WHERE `sourceType`= 1 AND `entry` IN (3117,26947);
INSERT INTO `disables` (`sourceType`,`entry`,`flags`,`params_0`,`params_1`,`comment`) VALUES
(1,  3117, 0, '', '', 'Obsolete quest: Etched Sigil'),
(1,  26947, 0, '', '', 'Obsolete quest: A Woodsman\'s Training);
```

**Comment by tkrokli** [comment]
Nice work, but I would enjoy it even more if your fixes had been compared with current TDB 703.00 content. For the orc starting area, I found an additional 15 quests to disable and a couple of missed quest givers.

**Comment by Keldo** [comment]
@tkrokli I am working on these using current 7-3 content so Im not getting where you are finding more quest unless you are looking outside of the initial start area which I have not begun to do, this is a collaborative effort so, let me know what you are doing so I can get in synch with your efforts.

**Comment by tkrokli** [comment]
OK, to be more precise about what I meant about comparing the fixes with current TDB content: I simply ran a query to see what was in the 2 tables `creature_queststarter` & `creature_questender` like this:
```sql
SELECT * FROM `creature_questender` WHERE `id` IN (3153,3155,3157,39214);
```
That in itself showed me that there were more than 2 class quests per each trainer, meaning that I would have to check the rest of the posted fix for the Valley of Trials. I then remembered that there should be more than 4 class trainers, even for the orcs, so I checked for mage trainer & warlock trainer and I found these 2:
```
3156   -- Nartok <Warlock Trainer>
39206  -- Acrypha <Mage Trainer>
```
I also looked up some further information regarding these trainers on wowhead, verifying when they were added to the game to see if it was worth looking for more quests. So I expanded the query to include those 2 NPCs, as well as throwing in the 3.3.5 troll priest trainer (BTW which is not spawned in TDB 703.00).
```sql
SELECT * FROM `creature_queststarter` WHERE `id` IN (3153,3155,3157,3707,39206,39214);
SELECT * FROM `creature_questender` WHERE `id` IN (3153,3155,3157,3707,39206,39214);
```
Hence my additions to the SQL based on what I found in both tables in my local copy of the TDB.
If you check findings in a DB from before commit 87b5c78, you will see that there is more to be removed than in your suggested fix. My main goal was to be sure to get rid of any obsolete quest data.

I also ran this query to check the quest names in TDB 703 so I had enough info for my commit:
```sql
SELECT `ID`,`LogTitle` FROM `quest_template` WHERE `ID` IN
(2383,3065,3082,3083,3084,3085,3087,3088,3089,3090,25138,25139,
 25140,25141,25142,25143,25144,25145,25146,25147,25148,25149,25150);
```
Of course, I could have elaborated with some `LEFT JOIN` or `RIGHT JOIN` queries to get a combined list of IDs + names of the quest givers and their quests in the same output, but I because I am generally slow at working out such queries, I preferred to just copy simple outputs to Notepad++ and copy-paste the info I needed to see the results. Feel free to play with SQL `JOIN` if you like. I did what I believe should be done.

**Comment by tkrokli** [comment]
Notice that the trainers in Northshire Valley are still quest givers, handing out [Join the Battle!](http://www.wowhead.com/search?q=Join_the_Battle) (9 variants):
- http://www.wowhead.com/quest=28780/join-the-battle (Ashley Blank, Hunter Trainer)
- http://www.wowhead.com/quest=28784/join-the-battle (Khelden Bremen, Mage Trainer)
- http://www.wowhead.com/quest=28785/join-the-battle (Brother Sammuel, Paladin Trainer)
- http://www.wowhead.com/quest=28786/join-the-battle (Priestess Anetta, Priest Trainer)
- http://www.wowhead.com/quest=28787/join-the-battle (Jorik Kerridan, Rogue Trainer)
- http://www.wowhead.com/quest=28788/join-the-battle (Drusilla La Salle, Warlock Trainer)
- http://www.wowhead.com/quest=28789/join-the-battle (Llane Beshere, Warrior Trainer)
- http://www.wowhead.com/quest=29080/join-the-battle (Marshal McBride)
- http://www.wowhead.com/quest=31143/join-the-battle (Bao, Monk Trainer)

**Comment by ViserionLIVE** [comment]
**-- Tauren starting area ( Mulgore )**

<details>

```sql

-- Low-level class quests have been removed in patch 7.0.3
-- Tauren starting area ( Mulgore )

-- The following NPCs are no longer quest givers:
UPDATE `creature_template` SET `npcflag` = `npcflag`& ~2 WHERE `entry` IN
( 3059,    -- Harutt Thunderhorn <Warrior Trainer>
 37737,    -- Sunwalker Helaku <Paladin Trainer>
 37724,    -- Seer Ravenfeather <Priest Trainer>
  3062,    -- Meela Dawnstrider <Shaman Trainer>
  3060,    -- Gart Mistrunner <Druid Trainer>
  3061,    -- Lanka Farshot <Hunter Trainer>
 63327);   -- Shoyu <Monk Trainer>

-- Rohaku Stonehoof (44927) no longer starts the class quests
DELETE FROM `creature_queststarter` WHERE `id`= 44927 AND `quest` IN
( 3091,    -- Quest: Simple Note (Warrior)
  3092,    -- Quest: Etched Note (Hunter)
  3093,    -- Quest: Rune-Inscribed Note (Shaman)
  3094,    -- Quest: Verdant Note (Druid)
-- 31165,    -- Quest: Calligraphed Note (Monk)
 27014,    -- Quest: Hallowed Note (Priest)
 27015);   -- Quest: Consecrated Note (Paladin)

-- These NPCs should no longer offer the following quests:
DELETE FROM `creature_queststarter` WHERE `id` = 3059 AND `quest`= 27020;  -- Harutt Thunderhorn <Warrior Trainer>, Quest: The First Lesson
DELETE FROM `creature_queststarter` WHERE `id` = 37737 AND `quest`= 27023; -- Sunwalker Helaku <Paladin Trainer>, Quest: The Way of the Sunwalkers
DELETE FROM `creature_queststarter` WHERE `id` = 37724 AND `quest`= 27066; -- Seer Ravenfeather <Priest Trainer>, Quest: Learning the Word
DELETE FROM `creature_queststarter` WHERE `id` = 3062 AND `quest`= 27027;  -- Meela Dawnstrider <Shaman Trainer>, Quest: Moonfire
DELETE FROM `creature_queststarter` WHERE `id` = 3060 AND `quest`= 27067;  -- Gart Mistrunner <Druid Trainer>, Quest: Verdant Note
DELETE FROM `creature_queststarter` WHERE `id` = 3061 AND `quest`= 27021;  -- Lanka Farshot <Hunter Trainer>, Quest: The Hunter's Path
DELETE FROM `creature_queststarter` WHERE `id` = 63327 AND `quest`= 31166; -- Shoyu <Monk Trainer>, Quest: Tiger Palm

-- and they also should no longer end these quests:
DELETE FROM `creature_questender` WHERE `id` = 3059 AND `quest` IN (3091,27020);   -- Harutt Thunderhorn <Warrior Trainer>, Quest: Simple Note & The First Lesson
DELETE FROM `creature_questender` WHERE `id` = 37737 AND `quest` IN (27015,27023); -- Sunwalker Helaku <Paladin Trainer>,   Quest: Consecrated Note & The Way of the Sunwalkers
DELETE FROM `creature_questender` WHERE `id` = 37724 AND `quest` IN (27014,27066); -- Seer Ravenfeather <Priest Trainer>,   Quest: Hallowed Note & Learning the Word
DELETE FROM `creature_questender` WHERE `id` = 3062 AND `quest` IN (3093,27027);   -- Meela Dawnstrider <Shaman Trainer>,   Quest: Rune-Inscribed Note & Primal Strike
DELETE FROM `creature_questender` WHERE `id` = 3060 AND `quest` IN (3094,27067);   -- Gart Mistrunner <Druid Trainer>,      Quest: Verdant Note & Moonfire
DELETE FROM `creature_questender` WHERE `id` = 3061 AND `quest` IN (3092,27021);   -- Lanka Farshot <Hunter Trainer>,       Quest: Etched Note & The Hunter's Path
DELETE FROM `creature_questender` WHERE `id` = 63327 AND `quest` IN (31165,31166); -- Shoyu <Monk Trainer>,                 Quest: Consecrated Note & Tiger Palm

-- Disable all the quests listed above
DELETE FROM `disables` WHERE `sourceType`= 1 AND `entry` IN (3091,27020,27015,27023,27014,27066,3093,27027,3094,27067,3092,27021,31165,31166);
INSERT INTO `disables` (`sourceType`,`entry`,`flags`,`params_0`,`params_1`,`comment`) VALUES
(1, 3091, 0, '', '', 'Obsolete quest: Simple Note'),
(1, 27020, 0, '', '', 'Obsolete quest: The First Lesson'),
(1, 27015, 0, '', '', 'Obsolete quest: Consecrated Note'),
(1, 27023, 0, '', '', 'Obsolete quest: The Way of the Sunwalkers'),
(1, 27014, 0, '', '', 'Obsolete quest: Hallowed Note'),
(1, 27066, 0, '', '', 'Obsolete quest: Learning the Word'),
(1, 3093, 0, '', '', 'Obsolete quest: Rune-Inscribed Note'),
(1, 27027, 0, '', '', 'Obsolete quest: Primal Strike'),
(1, 3094, 0, '', '', 'Obsolete quest: Verdant Note'),
(1, 27067, 0, '', '', 'Obsolete quest: Moonfire'),
(1, 3092, 0, '', '', 'Obsolete quest: Etched Note'),
(1, 27021, 0, '', '', 'Obsolete quest: The Hunters Path'),
(1, 31165, 0, '', '', 'Obsolete quest: Consecrated Note'),
(1, 31166, 0, '', '', 'Obsolete quest: Tiger Palm');

```

</details>

**Comment by tkrokli** [comment]
Thanks, I will have a look at that later. Looks good at a glance. 👍

**Comment by ViserionLIVE** [comment]
@tkrokli thanks to @Aokromes

**Comment by tkrokli** [comment]
Very well. 😄 I will see if I can get some of the latest additions committed at the end of this week.

**Comment by ViserionLIVE** [comment]
@tkrokli any news about this ?

**Comment by tkrokli** [comment]
@ViserionLIVE : Sorry for the delay. I have been more busy during office hours than usual and sometimes also not getting time to commit more updates during weekends. I will do my best to add Shadowglen updates during the next few days. I am also looking forward to see feedback regarding Anvilmar/Coldridge Valley.

**Comment by Keldo** [comment]
Anvilmar/Coldridge are broke and need to be re worked before we can get into removing unused Quest

**Comment by tkrokli** [comment]
OK, fair enough. I presume Coldridge Valley still uses 3.3.5 quests/scripting as a basis, as well as contains bugs in quest log handling.

**Comment by Keldo** [comment]
Coldridge Valley, everything just stands there and sometimes its absolutely impossible to kill anything, quest are out of order and cannot be completed. The entire area is just broke...

**Comment by Antarus26** [comment]
Any update on Coldridge Valley? Quests still seem to have some issues.

**Comment by tkrokli** [comment]
commit with minor fixes to Coldridge Valley: https://github.com/TrinityCore/TrinityCore/commit/5b3ffd44e1ef86ac2d12f0590489025064852dcd
If you still see issues, check if they are reported separately
or make a new report ticket using the issue template.

**Comment by Aokromes** [comment]
A lot where fixed, if there are some remains, open a new ticket writing affected quests.
